runner/src/server/services/upload/uploadService.ts

```diff
@@ -123,7 +124,17 @@ export class UploadService {
   }
 
   parsedDocumentUploadResponse({ res, payload }) {
-    const warning = payload?.toString?.();
+    const payloadString = payload?.toString?.();
+    const payloadJson = payloadString ? JSON.parse(payloadString) : undefined;
```

Comment:
I think you will need to wrap in try catch here. JSON.parse will throw if it's not valid JSON. You could possibly also check if the response header has content type of application/json? (not sure if you have access to it at this point)

I think you have dealt with this case below but just in case - I will note that FCDO does use text/plain and not JSON response. So updating this will break their services when they update the runner, since the runner will be parsing the responses differently. I'd recommend not introducing a breaking change. You can double check their API here https://github.com/UKForeignOffice/document-upload.